docs/design/contextual-tabs.md

```diff
@@ -0,0 +1,392 @@
+            },
+            {
+                "size": 32,
+                "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/WriteDataButton32x32.png"
+            },
+            {
+                "size": 80,
+                "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/WriteDataButton80x80.png"
+            }
+        ]
+    }
+    ```
+ 
+The following is the complete example of the JSON blob:
+
+```json
+'{
+  "actions": [
+    {
+      "id": "executeWriteData",
+      "type": "ExecuteFunction",
+      "functionName": "writeData"
+    }
+  ],
+  "tabs": [
+    {
+      "id": "CtxTab1",
+      "label": "Data",
+      "groups": [
+        {
+          "id": "CustomGroup111",
+          "label": "Insertion",
+          "icon": [
+            {
+                "size": 16,
+                "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/Group16x16.png"
+            },
+            {
+                "size": 32,
+                "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/Group32x32.png"
+            },
+            {
+                "size": 80,
+                "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/Group80x80.png"
+            }
+          ],
+          "controls": [
+            {
+                "type": "Button",
+                "id": "CtxBt112",
+                "actionId": "executeWriteData",
+                "enabled": false,
+                "label": "Write Data",
+                "superTip": {
+                    "title": "Data Insertion",
+                    "description": "Use this button to insert data into the document."
+                },
+                "icon": [
+                    {
+                        "size": 16,
+                        "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/WriteDataButton16x16.png"
+                    },
+                    {
+                        "size": 32,
+                        "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/WriteDataButton32x32.png"
+                    },
+                    {
+                        "size": 80,
+                        "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/WriteDataButton80x80.png"
+                    }
+                ]
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}'
+```
+
+## Register the contextual tab with Office with requestCreateControls
+
+The contextual tab is registered with Office by calling the [Office.ribbon.requestCreateControls](/javascript/api/office/office.ribbon?view=common-js&preserve-view=true#requestcreateconrtols-input-) method in either the function that is assigned to `Office.initialize` or with the `Office.onReady` method. For more about these methods and initializing the add-in, see [Initialize your Office Add-in](../develop/initialize-add-in.md).
```

Comment:
This could be a recommendation but it is not mandatory, it can be done at any time including after activation but only 1 time successfully.